nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -75,27 +76,45 @@ in
         };
       };
 
-      dbdserver = {
-        enable = mkEnableOption (lib.mdDoc "SlurmDBD service");
+      settings = mkOption {
+        type = settingsFormat.type;
 
-        dbdHost = mkOption {
-          type = types.str;
-          default = config.networking.hostName;
-          defaultText = literalExpression "config.networking.hostName";
-          description = lib.mdDoc ''
-            Hostname of the machine where `slurmdbd`
-            is running (i.e. name returned by `hostname -s`).
-          '';
-        };
+        description = mdDoc ''
+          Contents of `slurm.conf`.
+          An interactive configuration generator can be found [here](http://slurm.schedmd.com/configurator.html).
+        '';
 
-        storageUser = mkOption {
-          type = types.str;
-          default = cfg.user;
-          defaultText = literalExpression "config.${opt.user}";
-          description = lib.mdDoc ''
-            Database user name.
-          '';
-        };
+        example = literalExpression ''
+          settings = {
+            SlurmctldHost = "control";
+            nodeName = {
```

Comment:
The advantages that I see is that you can now (1) access the parameters for a node or a partition directly and use them in another place in your config (e.g. for CPUs or memory). (2) The attributes can be merged, when they are defined in separate locations. This is otherwise only possible if you define `nodeName`, `partitionName` separately (now also `nodeSet` exists).  (3) It makes the module easier to maintain and can automatically handle new features without the need to modify the module.

It may look over engineered at first glance, but I think it is more consistent to go all the way here.